pymc/tests/test_distributions_random.py

```diff
@@ -1698,6 +1718,34 @@ class TestOrderedProbit(BaseTestDistributionRandom):
         "check_rv_size",
     ]
 
+    @pytest.mark.parametrize(
+        "eta, cutpoints, sigma, expected",
+        [
+            (0, [-2.0, 0, 2.0], 1.0, (4,)),
+            ([-1], [-2.0, 0, 2.0], [2.0], (1, 4)),
+            ([1.0, -2.0], [-1.0, 0, 1.0], 1.0, (2, 4)),
+            ([1.0, -2.0, 3.0], [-2.0, 0, 2.0], [-1.0, -2.0, 5.0], (3, 4)),
+            ([[1.0, -1.0, 0.0], [-1.0, 3.0, 5.0]], [-2.0, 0, 1.0], [-1.0, -2.0, 5.0], (2, 3, 4)),
+            (
+                [[1.0, -2.0, 3.0], [1.0, 2.0, -4.0]],
+                [-2.0, 0, 1.0],
+                [[0.0, 2.0, -4.0], [-1.0, 1.0, 3.0]],
```

Comment:
A lot of these sigma are negative. We should test with valid sigma values (> 0)

Reply:
Also test with 2d cutpoints missing

Reply:
Hi @ricardoV94, I will change the `sigma` to all positive. 

Also, `cutpoints` should be always 1 dimension (to my understanding) as it represents (n-1) cut points of a categorical feature with n categories. I am not sure if there is any cases that needs 2d `cutpoints`.

Reply:
Our distributions, when possible, can always be "batched". That means we can arbitrarily increase the dimensionality of the distribution by adding parameters with more dimensions.

For example the categorical distribution is happy to take 2D, 3D, ... ND dimensional probability parameters.

The same should apply here if possible. See this issue where we are pursuing this for all multivariate distributions: https://github.com/pymc-devs/pymc/issues/5383

The reason why this is useful is vectorization. Specifying a (3, 3) shaped distribution with different cutpoints can be much more efficient than specifying 3 times a (3,) shaped distribution with the different cutpoints.

Reply:
This was exactly how this issue started by the way, just with the batching across sigma and eta, and fixed cutpoints. But it could have been the other way around.

Reply:
Hi yes, the `batch` dimension totally make sense. My initial thought was that for dealing with a large data set, `batch_size` should be managed in `pm.Data` (similar to DataLoader in pytorch). Although I have not checked `pm.Data` yet :)

Anyways, I will check the case of 2d cutpoints as well.

Reply:
Maybe now my crazy example from before makes more sense?

```python
OrderedProbit.dist(
  eta=np.zeros((5, 2)), 
  sigma=np.ones((2, 5, 2)), 
  cutpoints=[[-2, 0, 2], [-2, 0, 2]]
)  #shape should be (2, 5, 2, 4)
```

Reply:
Hi, it makes sense. I have added the tests for 2d `cutpoints` and positve `sigma`.

Already run `push`, but not sure why it has not updated in this PR:
https://github.com/danhphan/pymc/commit/9b311bf8b4d8fae93db2a085fb5be1e8c730c570